src/Microsoft.Extensions.CodeGenerators.Mvc/Templates/ControllerGenerator/MvcControllerWithContext.cshtml

```diff
@@ -171,7 +171,7 @@ namespace @Model.ControllerNamespace
 @{
     foreach (var property in relatedProperties.Values)
     {
-            @:ViewData["@(property.ForeignKeyPropertyNames[0])"] = new SelectList(_context.@property.EntitySetName, "@property.PrimaryKeyNames[0]", "@property.DisplayPropertyName", @(Model.ModelVariable).@property.ForeignKeyPropertyNames[0]);
+            @:ViewData["@(property.ForeignKeyPropertyNames[0])"] = new SelectList(_context.@property.EntitySetName, "@property.PrimaryKeyNames[0]", typeof(@property.EntitySetName).GetProperties().Where(p=>p.Name!= "@property.PrimaryKeyNames[0]").FirstOrDefault().Name, @(Model.ModelVariable).@property.ForeignKeyPropertyNames[0]);
```

Comment:
The problem with this is that it chooses the first property from the navigation that is not the primary key. 

Consider the model: 

```
public class Car 
{
        public int CarId { get; set; }
        public string Name { get; set; }
        public int ManufacturerId { get; set; }
        public virtual Manufacturer Manufacturer { get; set; }
}

public class Manufacturer
{
        public int ManufacturerId { get; set; }
        public bool IsCertified { get; set; }
        public string Name { get; set; }
        public virtual ICollection<Car> Cars { get; set; }
}

```
In this model if the Display property ends up being `IsCertified`, the drop down would consist of `true` and `false` options for each of the available `Manufacturer` entries.